query/src/storages/fuse/io/block_reader.rs

```diff
@@ -85,12 +86,20 @@ impl BlockReader {
         let arrow_fields = &self.arrow_table_schema.fields;
         let stream_len = self.file_len;
         let read_buffer_size = self.read_buffer_size;
+        let parquet_fields = metadata.schema().fields();
 
+        // read_columns_many_async use field name to filter columns
         let fields_to_read = self
             .projection
             .clone()
             .into_iter()
-            .map(|idx| arrow_fields[idx].clone())
+            .map(|idx| {
```

Comment:
the test `test_fuse_table_normal_case`
 failed because parquet is written with field names 'a', 'b' ...
while the block reader later read with field name 'id'